src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -1262,6 +1280,36 @@ private void updateCost() {
         }
     }
 
+    /**
+     * Check whether to activate Hybrid Updating Strategy (HUS)
+     */
+    private void checkHUS() {
+        if (!useHUS) {
+            return;
+        }
+
+        if (routeIteration == 1) {
+            // Count the number of overused nodes
+            long overUseCnt = 0;
+            for (RouteNode rnode : routingGraph.getRnodes()) {
+                if (rnode.isOverUsed()) {
+                    overUseCnt++;
+                }
+            }
+            HUSinitialCongested = (float) overUseCnt / numConnectionsToRoute > config.getHUSinitialCongestedThreshold();
+        }
+
+        if (HUSinitialCongested) {
+            float congestedConnRatio = (float) connectionsRoutedIteration / sortedIndirectConnections.size();
+            if (congestedConnRatio < config.getHUSactivateThreshold()) {
+                // Activate HUS: slow down the present cost growth and increase historical cost growth instead
```

Comment:
Since the present congestion multiplier and the historical congestion factor is set by the user, this statement may not always be true.

Perhaps we should have a check here that emits a warning if we're not actually slowing/increasing?